repoparse.py

```diff
@@ -0,0 +1,294 @@
+from dataclasses import dataclass
+from typing import Dict, List
+
+import sys
+
+from dulwich.objects import Commit, Blob
+from dulwich.patch import patch_filename, is_binary, gen_diff_header, unified_diff
+from dulwich.repo import Repo
+from os import getcwd, path
+
+from os import makedirs, system
+from shutil import rmtree
+
+from unidiff import PatchSet, UnidiffParseError
+
+from shared import flush_progress_bar, progress, finish_progress, group, s_get_content, s_get_lines, unwrap_bytes_gen_to_str
+
+import multiprocessing
+
+from treesitter_stage import load_treesitter
+from tree_sitter import Language, Parser
+
+@dataclass(init=True)
+class ParsedCommitInfo:
+
+    # Only single parent is allowed
+    parent_sha: str
+    patch_set: Dict[str, str]
+    langs: Dict[str, str]
+    ids: Dict[str, str]
+
+
+@dataclass
+class ParsedCommits:
+
+    commit_data_by_sha: Dict[str, ParsedCommitInfo]
+
+    def __getitem__(self, item: str) -> ParsedCommitInfo:
+        return self.commit_data_by_sha.get(item)
+
+    def __setitem__(self, key: str, value: ParsedCommitInfo):
+        self.commit_data_by_sha[key] = value
+
+
+@dataclass
+class ParsedRepo:
+
+    commits_by_author: Dict[str, ParsedCommits]
+
+    def __getitem__(self, item: str):
```

Comment:
as an option you can use property - https://stackoverflow.com/questions/2627002/whats-the-pythonic-way-to-use-getters-and-setters